sequencer/src/block.rs

```diff
@@ -81,6 +88,10 @@ pub struct Header {
     pub block_merkle_tree_root: BlockMerkleCommitment,
     /// Root Commitment of `FeeMerkleTree`
     pub fee_merkle_tree_root: FeeMerkleCommitment,
+    /// Account (etheruem address) of builder
+    pub builder_address: Option<FeeAccount>,
+    pub builder_signature: Option<Signature>,
+    pub builder_fee_amount: Option<FeeAmount>,
```

Comment:
These three fields have to be present together or not at all, so it would be cleaner to have `fee_info: Option<(FeeAccount, Signature, FeeAmount)>`. Maybe we can make a struct like
```rust
struct FeeInfo {
  account: FeeAccount,
  signature: Signature,
  amount: FeeAmount,
}
```

Reply:
Humm they can't *really* be there "together or none at all" ... can they? I have to be able to construct the type w/out the signature, sign it, and then add the signature. Wouldn't it have to be something like:
```
FeeInfo {
  account: FeeAccount,
  signature: Option<Signature>,
  amount: FeeAmount
}
Header {
  fee_info: Option<FeeInfo>,
  ..snip..
}
```
Or maybe to think of it another way... Signature isn't really the signature of fee stuff, its the signature of the entire Header. So I might prefer this:

```
FeeInfo {
  account: FeeAccount,
  amount: FeeAmount
}
Header {
  fee_info: Option<FeeInfo>,
  signature: Option<Signature>,
  ..snip..
}
```
But then that makes me wonder if we can't just always have the FeeInfo and so it would be more like this:
```
FeeInfo {
  account: FeeAccount,
  amount: FeeAmount
}
Header {
  fee_info: FeeInfo,
  signature: Option<Signature>,
  ..snip..
}
```

Reply:
Huh yeah I kind of like that last idea. `FeeInfo` could just be default for the genesis with no signature.